### ISSUE TITLE
letzter Schliff

### ISSUE THREAD

**Issue by TSchiefer** [TSchiefer/CAS_InfEng_DB_DWH#4]
Ich erstelle jetzt noch mal ein  letztes Issue, damit's übersichtlich bleibt auf den letzten Metern.

**Comment by TSchiefer** [comment]
Habe im Commit abbedc4b9aa5426ba8b312085451931124f1294b nochmal meinen Teil des Reports aktualisiert. Wäre super, wenn Ihr es kurz durchsehen könntet. Insbesondere erkläre ich jetzt, warum wir nicht die staging-area Datenbank im Datenflow integrieren, sage aber auch, dass wir eigentlich schon wissen, wie wir das umsetzen würden - allein es fehlt die Zeit.

Ausserdem habe ich das ZIP-File-Directory in Commit cb32490f291ea4f44d3828f325d3bd9139a2fc2e so umsortiert, dass Unterverzeichnisse dort den Kapiteln im Report entsprechen. Hoffe, das ist so in Eurem Sinne.

**Comment by TSchiefer** [comment]
@Eckeau Schaue gerade die Queries durch:

kann es sein, dass es bei der zweiten Abfrage besser wäre, wie in der ersten Abfrage "SELECT SUM(f.GEWINN)" zu schreiben? Es kommt natürlich das gleiche raus, da der Kunde "Company A" nur einmal Kaffee gekauft hat. Aber wenn er das zweimal gemacht hätte, würdest Du zwei Reihen als Antwort bekommen und nicht den Gesamtgewinn mit Kaffee mit Kunde "Company A". Glaube ich zumindest...

**Comment by FischerAndi** [comment]
Ich habe meinen Report-Beitrag gepushed (4 Seiten) und auch zwei Graphiken abgelegt.
Ich weiss nicht, was ich noch schreiben könnte, das lesenswert wäre :-)

Da ich nur Libre Office habe, könnte meine Formatierung etwas "buggy" sein z.B. Titelnummerierung.

**Comment by TSchiefer** [comment]
Danke, schaut gut aus und ich kann es auch gut lesen.

Dein ERM-Modell ist natürlich kaum entzifferbar, wenn es in das Word-Dokument eingebunden ist. Aber es wird ja im zip-File mit drinnen sein.

Könntet Ihr beide bitte nochmal sicherstellen, dass alle Files, die Ihr im zip-File haben wollt, auch in der richtigen Version im entsprechenden Verzeichnis im Ordner "Dateien_für_ZIP-File/" sind?

**Comment by TSchiefer** [comment]
@Eckeau bin jetzt durch Deine Queries durch. Bis auf das oben erwähnte, schaut es für mich richtig aus, und das Level ist auch durchaus passend, denke ich. Danke Dir!

Ich fange dann mal an, die Reporte zu konsolidieren und schicke Euch dann den zusammengesetzten Report, sobald er fertig ist, zur letzten Kontrolle. Das zip-File kommt dann natürlich auch.

Wenn Ihr noch was hochladet, bitte in diesem Issue einen Kommentar schreiben, sonst bekomme ich es nicht mit.

**Comment by Eckeau** [comment]
@TSchiefer  Ja, du hast recht. Ich passe die Query an und mache einen neuen Screenshot. Soll ich eine neue Version des ganzen Word Dokuments anlegen oder nur den Screenshot machen und Du ersetzt das dann?

**Comment by TSchiefer** [comment]
oh, und wenn Euch was für Kapitel 8 einfällt, bitte einfach drauf losschreiben :)

Ich schreibe ein paar Zeilen zur Arbeit mit `git`. Bernd, könntest Du bitte das über die Probleme mit den Fremdschlüsseln schreiben?

Und zu Deinem Kommentar gerade: ich kann den Screenshot ersetzen, wenn Du ihn hochlädst.

**Comment by FischerAndi** [comment]
Ich habe alles im Repository in der letzten Version.

**Comment by Eckeau** [comment]
Screenshot ist hochgeladen.

**Comment by Eckeau** [comment]
Bin mir nicht sicher, ob ich das Fremdschlüssel-Problem gut genug verstanden habe, um es zu beschreiben. Ich schaue mal, ob ich noch was herausfinde.

**Comment by Eckeau** [comment]
Noch eine Anmerkung: ich habe den Font für den Text bei mir auf 12 herunter gesetzt. 14 erschien mir sehr gross und das Inhaltsverzeichnis ist auch 12er Grösse.

**Comment by Eckeau** [comment]
Ok, dann folgt als nächster Kommentar mein Textvorschlag für den Anfang von Kapitel 3.

**Comment by Eckeau** [comment]
Zunächst ein paar Worte zu unserem Umgang mit der staging-area: Zur effizienteren Arbeitsaufteilung haben wir die Aufgaben der Einrichtung der staging-area und der Erzeugung des multidimensionalen Modells getrennt. Aus Gründen der Einfachheit und der Möglichkeit, schneller mit der Arbeit zu beginnen, haben wir die Daten für die verschiedenen Transformationen (Reparatur der korrupten Daten, Befüllen des multidimensionalen Modells, ...) aus csv-Dateien eingelesen und teils auch wieder als csv-Dateien rausgeschrieben. Die staging-area bestand somit letztlich aus dem Filesystem.

Aus zeitlichen Gründen haben wir darauf verzichtet, dies noch so umzustellen, dass die staging-area in Form einer Datenbank implementiert wurde, d.h. das Einlesen – bis auf den einen Schritt zur Befüllen der staging-area Datenbank – immer aus Datenbanktabellen vorgenommen werden würde. 

Wir wollen aber gerne demonstrieren,  wie theoretisch die Integration der Daten über die Datenbank ausgesehen hätte. Dazu haben wir die staging-area Datenbank erstellt und mit den notwendigen Tabellen befüllt. Nur eben die Integration in den Daten-Flow ist nicht gegeben.

Es folgt eine kurze Erläuterung, wie unsere staging-area-Datenbank erzeugt und befüllt wird.

**Comment by Eckeau** [comment]
Hast Du schon integriert oder können wir noch Änderungen an unseren Kapiteln vornehmen?

**Comment by TSchiefer** [comment]
Ich bin gerade beim Integrieren. Ich habe dazu meinen Teil des Reports umbenannt in "Report_Projekt_Modul_DB_DWH_Tobi.docx" und füge Eure Teile mit Copy-Paste ein.

Wenn Ihr noch was ändern wollt an Euren Teilen, könnt Ihr das gern machen. Ladet dann einfach Euer neues docx-File hoch und schreibt im Kommentar, wo die Änderungen passiert sind. Dann kann ich diesen Teil einfach erneut übertragen.

**Comment by TSchiefer** [comment]
Der erste Satz des gerade geschriebenen Kommentars hätte lauten sollen:
'Ich bin gerade beim Integrieren. Ich habe dazu meinen Teil des Reports umbenannt in "Report_Projekt_Modul_DB_DWH_Tobi.docx" und füge Eure Teile mit Copy-Paste in ein auf meinem Teil basierendes File mit dem ursprünglichen Namen "Report_Projekt_Modul_DB_DWH.docx" ein.'

**Comment by TSchiefer** [comment]
Danke für Deine Änderungsvorschläge für den Beginn von Kapitel 3, ist eingebaut

**Comment by TSchiefer** [comment]
Im Commit cbb5131ea2f8f43635a516526d96b3a042f039ee habe ich Andi's Teil und meinen zusammengebaut.

@FischerAndi ich habe dabei:
1. die Anführungszeichen «» zu denen hier "" gemacht (ich dachte mir, ich muss es in eine Richtung machen, und ich wusste nicht, wie ich es standardmässig einstellen kann, dass es diese Anführungszeichen bei mir macht)
2. Die Bilder mit Captions versehen und im Text Links darauf gemacht.

Hoffe, das passt so für Dich.

**Comment by FischerAndi** [comment]
Passt so und mach es wie Du es für richtig hälst. Es soll ja einheitlich aussehen.

**Comment by TSchiefer** [comment]
Ich habe gerade festgestellt, dass mir meine etwas veraltete Word-Version nicht die Möglichkeit bietet, das Dokument als pdf zu exportieren, sondern lediglich als solches zu speichern. Deshalb funktionieren die Links vom TOC zu den Kapiteln und auch die Links zu den im Text referenzierten Bildern nicht.

Hat einer von Euch die Export-Funktion in Word bzw. LibreOffice? Funktioniert das dann mit den Links?

**Comment by FischerAndi** [comment]
Libre Office funktioniert zum Teil. TOC zu den Kapiteln funktioniert nicht. Kann man vielleicht einstellen.
Versuche weiter.

Übrigens: Ich habe gesehen, dass in meinem Teil noch die Nummerierung der Unterkapiteln noch nicht stimmt. Es gibt zwei Kaptiel A.

**Comment by Eckeau** [comment]
Soll ich das Bild als PDF oder als PNG hochladen?

**Comment by TSchiefer** [comment]
ich persönlich würde pdf bevorzugen

**Comment by Eckeau** [comment]
Habe nun einfach beides ins Verzeichnis gelegt.

**Comment by Eckeau** [comment]
Sagt mir Bescheid, wenn ich was mit der Original Windows Word Version machen soll. PDF speichern geht sicher

**Comment by Eckeau** [comment]
@TSchiefer  Die Änderungen, die ich an meinem Teil noch hätte wären minimal. Es geht nur darum bei den SQL Statements bei den CREATE TABLE Befehlen jeweils noch ein IF NOT EXISTS einzufügen. Da war ich etwas schlampig. Ist aber auch nicht tragisch, wenn das fehlt.

**Comment by TSchiefer** [comment]
Zwei Commits:
- b9a155976afb924d61b59daa3538bc3547bc53c9: Hinzunahme der beiden Files "01_Aenderung_ERM_Dialekt/Northwind-ERM-Schema-v3.png" und "05_Konzipierung_Sternschema/ERM Star Schema-v2.pdf" zum zip-File
- Mit 3803435e1a3402290f0dd05c819671e471c70034 hat der Report ("Report/Report_Projekt_Modul_DB_DWH.docx") jetzt alle Teile beisammen. Ich habe auch ein paar Worte in Kapitel 8 verloren. Schreibt ruhig noch was dazu, wenn Ihr möchtet. 

Aber verkünsteln brauchen wir uns ja auch nicht, ich denke, bestehen werden wir schon damit.

**Comment by TSchiefer** [comment]
@Eckeau ich denke auch, dass wir die SQL-Statements so belassen können, wie sie sind.

**Comment by TSchiefer** [comment]
@Eckeau wenn wir uns alle einig sind, dass es so passt, wäre es super, wenn Du den Export ins pdf-Format vornehmen könntest und das File dann hochladen könntest. Dann kann einer von uns unserem Dozenten die erlösende Email schicken.

**Comment by FischerAndi** [comment]
Wouw. Sieht alles super aus.

**Comment by Eckeau** [comment]
PDF ist hochgeladen. Bitte nochmal durchchecken, ob alles mitgekommen ist. Für mich sieht es gut aus.

31 Seiten!!! Da wünsche ich dem Kollegen Aebi viel Spass beim Lesen ;-)

Auch wenn nicht alles perfekt stimmen sollte kann er uns da unmöglich den Segen verweigern. Danke Euch beiden für die Zusammenarbeit.

**Comment by Eckeau** [comment]
@TSchiefer  Doch noch ein last minute change von mir. Neues PDF ist hochgeladen.

Ich habe nun noch erfolgreich mit Foreign-Key Constraints auf der Faktentabelle arbeiten können. Allerdings darf ich die Zeit-Dimension nicht einbeziehen (keine Ahnung warum). Ich habe die Befehle im Dokument noch ergänzt. Im Code für das zip-File lasse ich es mal weg.

**Comment by Eckeau** [comment]
Wer schickt die Mail?

**Comment by TSchiefer** [comment]
Habe gerade gesehen, dass leider die Links im pdf doch nicht funktionieren und ich wwürde gern noch eben was ausprobieren. Glaube, das müsste klappen.

Ich sag dann gleich noch Becheid.

**Comment by Eckeau** [comment]
@TSchiefer  Ich bin unterwegs und der Strom wird knapp. Daher wäre ich froh, wenn Du den Versand machen kannst.

**Comment by Eckeau** [comment]
Welche Links gehen denn nicht? Bei mir sieht es ok aus.

**Comment by TSchiefer** [comment]
Ich glaube das ist ein Problem bei mir mit dem Apple Programm "Preview". Ich kann beim TOC nicht auf die Links klicken. Die Links zu den Bildern funktionieren aber, alles paletti.

Ich schicke dann das Email in der nächsten halben Stunde ab :)

**Comment by Eckeau** [comment]
Stimmt, TOC geht bei mir auch nicht. Habe aber auch keine Konfigurations-Option dafür beim Speichern gefunden.

**Comment by TSchiefer** [comment]
Ist ja auch kein Problem, die Seiten sind ja jetzt richtig (sorry, dass ich das Update vom TOC vorher vergessen hatte)

**Comment by TSchiefer** [comment]
Submitted ist's, juchhu!

**Comment by Eckeau** [comment]
Super, Danke!

**Comment by TSchiefer** [comment]
eh, Ihr habt die Email schon auch bekommen, oder?

Ich hab's von dem ZHAW-Outlook abgeschickt, nicht, dass ich da was Falsches gedrückt hab oder sonstwas schiefgelaufen ist...

**Comment by FischerAndi** [comment]
Vielen Dank Euch beiden. 
Ich konnte mich gerade nicht auf dem ZHAW-Mail anmelden, wegen Wartungsarbeiten.

**Comment by FischerAndi** [comment]
Jetzt ist es gegangen und das Mail ist angekommen. Super.

**Comment by TSchiefer** [comment]
Vielen Dank auch an Euch beide und noch einen schönen Abend

**Comment by FischerAndi** [comment]
Falls ihr es noch nicht gesehen habt: Wir haben den Leistungsnachweis erbracht.
Gratulation, das haben wir gut gemacht ;-)

Gruss

Andi

**Comment by Eckeau** [comment]
Danke für den Hinweis und Danke nochmal für den Einsatz.
Er hat schon seine Ansprüche der liebe Dani Aebi, aber so lernt man auch was.
